docs/dictionary-cache.md

```diff
@@ -0,0 +1,51 @@
+# Dictionary cache 
+
+Percona XtraBackup is based on how [`crash recovery`](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_crash_recovery) works. Percona XtraBackup copies the InnoDB data files, which results in data that is internally inconsistent; then the `prepare` phase performs crash recovery on the files to make a consistent, usable database again.
+
+The `--prepare` phase has the following operations:
+
+* Applies the redo log
+* Applies the undo log
+
+As a physical operation, the changes from the redo log modifications are applied to a page. In the redo log operation, there is no concept of row or transaction. The redo apply operation does not make the database consistent with a transaction. An uncommitted transaction can be flushed or written to the redo log by the server. Percona XtraBackup applies changes recorded in the redo log.
+
+Percona XtraBackup physically modifies a specific offset of a page within a tablespace (IBD file) when applying a redo log. 
+
+As a logical operation, Percona XtraBackup applies the undo log on a specific row. When the redo log completes, XtraBackup uses the undo log to roll back changes from uncommitted transactions during the backup.
+
+## Undo log
+
+There are two types of undo log records:
+
+* INSERT
+* UPDATE
+
+An undo log record contains a `table_id`. Percona XtraBackup uses this `table_id` to find the table definition, and then uses that information to parse the records on an index page. The transaction rollback reads the undo log records and applies the changes. 
+
+After initializing the data dictionary engine and the data dictionary cache, the storage engine can request the `table_id` and uses this ID to fetch the table schema. An index search tuple (key) are created from the table schema and key fields from the undo log record. The server finds the record using the search tuple (key) and performs the undo operation.
```

Comment:
An index search tuple (key) is created..

Reply:
fixed.